core/src/error/suggest.rs

```diff
@@ -0,0 +1,49 @@
+//! Suggest existing symbols that are similar to what the user typed in case of various "symbol not
+//! found" errors, such as [super::EvalError::UnboundIdentifier] or
+//! [super::EvalError::FieldMissing].
+//!
+//! The current implementation uses the normalized Damereau-Levenshtein edit distance to find the
+//! closest match.
+
+use strsim::normalized_damerau_levenshtein;
+
+/// The minimum similarity between the user's input and an existing symbol for the symbol to be
+/// considered a serious candidate. The current threshold is rather low, because short words with
+/// edit distance 1 such as `bar` and `bare` might have a similarity which isn't that high.
+pub const MIN_SIMILARITY: f64 = 0.60;
+
+/// Find the closest match to the user input in the given list of symbols, to be used as a
+/// suggestion. If there is an exact match (modulo casing), it is returned immediately. Otherwise,
+/// the symbol with the highest normalized Damerau-Levenshtein similarity is returned, as long as
+/// the similarity is above the fixed threshold [MIN_SIMILARITY].
+pub fn find_best_match<'syms, 'input, S, I>(
```

Comment:
I would rather suggest a few best matches rather than the best one. But I also don't have any proof that it's superior, so I'll leave it to your judgment.

Reply:
I _think_ Nix, rust and others generally provide only one matches, do they? I'm not sure. I don't have anything against providing a few ones, yeah.

Reply:
I'm leaving one suggestion for now, and let multiple suggestion for future work, in order to not delay even more 1.3.

Reply:
Nix at least provides several (but I'm the one who wrote it, so…). But yeah, it's certainly not worth blocking anything on this. We can see in the future whether it seems to become an annoyance or not.